functions/api/dashboard.js

```diff
@@ -63,23 +85,51 @@ async function getDashboardDataForBranchUser(branchId) {
     var dashboardData = {}
     dashboardData[constants.BRANCH] = branchId
     const branchRef = db.collection(constants.BRANCHES).doc(branchId);
-    const productsBelowThresholdPromise = getProductsUnderThreshold(branchRef)
+    const productsBelowThresholdPromise = getProductsBelowThreshold(branchRef)
     const recentTransactionsPromise = getRecentTransactions(branchRef)
     const pendingRequestsPromise = getPendingRequests(branchRef)
 
-    dashboardData[constants.PRODUCTS_BELOW_THRESHOLD] = await productsBelowThresholdPromise
+    var inventoryData = await productsBelowThresholdPromise
     dashboardData[constants.RECENT_ACTIVITY] = await recentTransactionsPromise
     dashboardData[constants.PENDING_REQUESTS] = await pendingRequestsPromise
+
+    dashboardData[constants.TOTAL_PRODUCTS_IN_INVENTORY] = inventoryData["totalProductsInInventory"]
+    dashboardData[constants.TOTAL_PRODUCTS_BELOW_THRESHOLD] = inventoryData["totalProductsBelowThreshold"]
+    dashboardData[constants.PRODUCTS_BELOW_THRESHOLD] = inventoryData["productList"]
     return dashboardData
 }
 
 /**
  * Method to retrieve products under threshold for a given branch
  * @param {*} branchId 
  */
-async function getProductsUnderThreshold(branchRef) {
-    const products = await getInventory(branchRef, branchRef.id, true)
-    return products.inventory
+async function getProductsBelowThreshold(branchRef) {
+    const allProductsInventory = await getInventory(branchRef, branchRef.id, false)
+    const products = allProductsInventory.inventory
+    const totalProductsInInventory = products.length
+    const productsBelowThreshold = []
+    
+    var responseCount = 0;
+    var totalProductsBelowThresholdCount = 0;
+
+    products.forEach(product => {
+        if (product[constants.IS_BELOW_THRESHOLD] === true) {
+            totalProductsBelowThresholdCount = totalProductsBelowThresholdCount + 1
+            responseCount = responseCount + 1
+            if(responseCount <= 5) {
+                productsBelowThreshold.push(product)
```

Comment:
what you are doing here is limiting the response to 5 right? In that case , once after 5 count is reached, dont you want to get out of the loop. Why to unnecessarily loop for all products

Reply:
Because I need to count totalProducts that are below threshold.